packages/quick_actions/quick_actions_android/android/src/main/java/io/flutter/plugins/quickactions/QuickActions.java

```diff
@@ -0,0 +1,183 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.quickactions;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ShortcutInfo;
+import android.content.pm.ShortcutManager;
+import android.content.res.Resources;
+import android.graphics.drawable.Icon;
+import android.os.Build;
+import androidx.annotation.ChecksSdkIntAtLeast;
+import androidx.annotation.NonNull;
+import io.flutter.plugins.quickactions.Messages.AndroidQuickActionsApi;
+import io.flutter.plugins.quickactions.Messages.FlutterError;
+import io.flutter.plugins.quickactions.Messages.Result;
+import io.flutter.plugins.quickactions.Messages.ShortcutItemMessage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+final class QuickActions implements AndroidQuickActionsApi {
+  protected static final String EXTRA_ACTION = "some unique action key";
+
+  private final Context context;
+  private Activity activity;
+
+  QuickActions(Context context) {
+    this.context = context;
+  }
+
+  void setActivity(Activity activity) {
+    this.activity = activity;
+  }
+
+  public Activity getActivity() {
+    return this.activity;
+  }
+
+  @ChecksSdkIntAtLeast(api = Build.VERSION_CODES.N_MR1)
+  boolean isVersionAllowed() {
+    // We already know that this functionality does not work for anything
+    // lower than API 25 so we chose not to return error. Instead we do nothing.
```

Comment:
This comment doesn't make sense here with the restructuring; it belongs on the `if` that calls it.